src/main/kotlin/org/rust/ide/intentions/SimplifyBooleanExpressionIntention.kt

```diff
@@ -0,0 +1,122 @@
+                    ANDAND, OROR -> leftSimplifiable // short-circuit operations
+                    else -> false
+                }
+            }
+            is RustUnaryExprElement ->
+                when (psi.operatorType) {
+                    EXCL -> psi.expr?.let { isSimplifiableExpression(it) } ?: false
+                    else -> false
+                }
+            is RustParenExprElement ->
+                isSimplifiableExpression(psi.expr)
+            else ->
+                false
+        }
+    }
+
+    override fun invoke(project: Project, editor: Editor, ctx: Context) {
+        val (expr) = ctx;
+        val value = calculateExpression(expr) ?: return
+        expr.replace(RustPsiFactory(project).createExpression("" + value))
+    }
+
+    private fun calculateExpression(expr: RustExprElement): Boolean? {
+        return when (expr) {
+            is RustBinaryExprElement ->
+                when (expr.operatorType) {
+                    ANDAND -> {
+                        val rhs = expr.right ?: return null
+                        val leftValue = calculateExpression(expr.left) ?: return null
+                        if (!leftValue)
+                            return false
+                        val rightValue = calculateExpression(rhs) ?: return null
+                        leftValue && rightValue
+                    }
+                    OROR -> {
+                        val rhs = expr.right ?: return null
+                        val leftValue = calculateExpression(expr.left) ?: return null
+                        if (leftValue)
+                            return true
+                        val rightValue = calculateExpression(rhs) ?: return null
+                        leftValue || rightValue
+                    }
+                    XOR -> {
+                        val rhs = expr.right ?: return null
+                        val leftValue = calculateExpression(expr.left) ?: return null
+                        val rightValue = calculateExpression(rhs) ?: return null
+                        leftValue xor rightValue
+                    }
+                    else -> null
+                }
+            is RustUnaryExprElement -> {
+                when (expr.operatorType) {
+                    EXCL -> expr.expr?.let { calculateExpression(it)?.let { !it } }
+                    else -> null
+                }
+            }
+            is RustParenExprElement ->
+                calculateExpression(expr.expr)
+            is RustLitExprElement -> {
+                if (expr.`false` != null)
+                    return false
+                if (expr.`true` != null)
+                    return true
+                null
+            }
+            else -> null
+        }
+    }
+
+    val RustUnaryExprElement.operator: PsiElement
+        get() = requireNotNull(node.findChildByType(UNARY_OPS)) { "guaranteed to be not-null by parser" }.psi
+
+    val RustUnaryExprElement.operatorType: IElementType
```

Comment:
What do you think about introducing `enum class UnaryOperator`? It'd give us a bit more typesafety in comparison with the `IElementType`.